solidity/contracts/Gravity.sol

```diff
@@ -184,6 +184,9 @@ contract Gravity is ReentrancyGuard {
 			abi.encodePacked("\x19Ethereum Signed Message:\n32", _theHash)
 		);
 		(address recAddr, ) = ECDSA.tryRecover(messageDigest, _v, _r, _s);
+
+		//in some cases ecrecover returns empty address.
+		require(recAddr != address(0), "ecrecover empty address");
```

Comment:
you can also fetch the message from the tryRecover function, you can use it instead if "ecreover emtpty address", becuase there are several cases where the function will return zero address but with different messsage